doc/commitlogs.md

```diff
@@ -0,0 +1,51 @@
+## M3DB Commit Logs
+
+### Integrity Levels
+
+There are two integrity levels available for M3DB commit logs:
+
+- **Synchronous:** all write operations must synchronously finish writing an entry in the commit log to complete.
+- **Buffered:** all write operations must enqueue to a buffer an entry in the commit log to complete.
+
+Depending on the data loss requirements consumers can choose either integrity level.
+
+### Properties
+
+Commit logs will be stamped by the start time, aligned and rotated by the block size.  To restore data for an entire block you will require the commit logs from the previous block size period, the current block size period and the future block size.  This is to account for the buffer past data and the buffer future data.
```

Comment:
Haven't gone through the rest of the diff yet, but seems like this means we need to weave data from three different commit log files to bootstrap a block size period? Also implies we need to always keep three commit log files around.

How about keeping two commit logs open instead, one for the current block period and one for the next block period, and don't close the current one until it's sufficiently old that no writes should be going into it (e.g., t0 + block size + `bufferPast` where t0 is the beginning of the current block period)? This way we only need to read from one commit log for one block period during bootstrapping.